addons/account_update_tax_tags/wizard/account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,169 @@
+                  JOIN account_move_line_account_tax_rel amlatr 
+                    ON amlatr.account_move_line_id = aml.id
+             -- Handle possible children_tax_ids
+             LEFT JOIN account_tax_filiation_rel atf 
+                    ON atf.parent_tax = amlatr.account_tax_id
+                  JOIN account_tax tax 
+                    ON tax.id = COALESCE(atf.child_tax, amlatr.account_tax_id)
+                  JOIN account_tax_repartition_line atrl
+                    -- the repartition line to join depends of the move_type
+                    -- also, as this is the base line query, we only join for the base repartition type
+                    ON
+                        atrl.repartition_type = 'base'
+                        AND (
+                            -- invoice type doc
+                            (move.move_type IN ('in_invoice','out_invoice', 'in_receipt', 'out_receipt') AND atrl.invoice_tax_id = tax.id)
+                            OR
+                            -- refund type doc
+                            (move.move_type IN ('in_refund','out_refund') AND atrl.refund_tax_id = tax.id) 
+                            OR
+                            -- entry type doc: depends on the tax type
+                            -- for base line:
+                            -- balance < 0 and sale tax --> invoice
+                            -- balance > 0 and sale tax --> refund
+                            -- balance > 0 and purchase tax --> invoice
+                            -- balance < 0 and purchase tax --> refund
+                            -- impossible to decide for balance at 0 --> invoice by default but it probably doesn't even matter
+                            (
+                                move.move_type = 'entry' 
+                                AND (
+                                    (tax.type_tax_use = 'sale'
+                                    AND (
+                                        aml.balance <= 0 AND atrl.invoice_tax_id = tax.id
+                                        OR aml.balance > 0 AND atrl.refund_tax_id = tax.id 
+                                    ))
+                                    OR
+                                    (tax.type_tax_use = 'purchase'
+                                    AND (
+                                        aml.balance >= 0 AND atrl.invoice_tax_id = tax.id
+                                        OR aml.balance < 0 AND atrl.refund_tax_id = tax.id
+                                    ))
+                                )
+                            ) 
+                        ) 
+             -- LEFT to allow keeping all aml even those without relation. The NULL will symbolize no relation
+             LEFT JOIN account_account_tag_account_tax_repartition_line_rel rep_tags ON rep_tags.account_tax_repartition_line_id = atrl.id
+                 WHERE move.id != ALL(%(excluded_move_ids)s)
```

Comment:
several reason and none of them are related to performance:
- this allowed to directly pass a tuple instead of a list to the execute function
- `NOT IN ()` results in a crash and would require to handle that part of the query dynamically
- if for any reason we get `NOT IN (..., NULL, ...)` it would result in undefined behavior. (But this point is just me trying to "flex", NULL shouldn't happen :D )

Also, if we don't exclude any move, we might just remove this condition.